clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -464,6 +465,11 @@ public static ByteBuffer toBuffer(Send send) {
         return exceptionCauseClass.cast(exception.getCause());
     }
 
+    public static void assertFutureThrowsException(Future<?> future, ApiException expectedException) {
+        ApiException receivedException = assertFutureThrows(future, expectedException.getClass());
+        assertEquals(expectedException.getMessage(), receivedException.getMessage());
```

Comment:
Shouldn’t the concrete exception class also be checked?

Reply:
Thanks for pointing that out. I have fixed this.